### PR TITLE
Working with Styled-components in React 

### LINKED ISSUE #2465: Working with Styled-components in React
We ask that you please be patient as our team works through approving and publishing all articles/tutorials in a timely manner. Allow 1-3 days for a topic to be reviewed and/or approved - allow 3-7 days for an articles to be reviewed and/or published.

Be sure to visit our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for tools, resources, and example articles to go over.

**NOTE: (Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**

**NOTE: We tend to stray away or tend not to publish reviews/comparisons of commercial product offerings.**

*We're looking for the first 2-3 paragraphs of the article that appropriately summarize what your article will be about.*

### Proposed title of article
Working with styled-components in React

### Introduction paragraph (2-3 paragraphs):
If you have worked with React in developing your applications, you will agree with me that adding style to your Components is super cool when using styled-components feature. In any case you have not used React or styled-components, do not worry because this article is for you.

Styled-components allows you to write CSS in your JavaScript application and we will leverage this to build our simple website.
In this tutorial, you will learn an overview of React and what styled-components is and why use it. In the end. we will build a simple web page in React and add styles to it using styled-components.

 
### Key takeaways: 
1. What is React.
2. What is styled-components.
3. Advantages of using styled-components.
4. Creating and styling a simple web-page using styled-components.

### References: 
N/A

### Templates to use as guides
- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

### PR DESCRIPTION
**Pull request by OMONDIELLY**
This fixes #2465 

### Article checklist

**ATTENTION: In your PR - add the ISSUE (topic form) # that this PR corresponds to. You can do this by also adding a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.**
      
      
**NOTE: (Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**

**If you have not already please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve our overall technical writing.**

#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker? We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**

### CODE REVIEW

**Comment by hectorkambow** [comment]
@OMONDIELLY 
Good afternoon and- thank you for submitting your article for publishing. It has been entered into our queue and should be reviewed (and published pending any changes) as soon as a peer reviewer is finished with the ones in the queue before it.  🚀

**Comment by lalith1403** [comment]
@OMONDIELLY, can you check with the SVG files in the repository. Kindly check if they are required.

**Comment by hectorkambow** [comment]
@OMONDIELLY 

cc @lalith1403  

Just wanted to get a status update on this PR - see if it was ready - as Im trying to clear our the queue.

**Comment by OMONDIELLY** [comment]
> @OMONDIELLY
> 
> cc @lalith1403
> 
> Just wanted to get a status update on this PR - see if it was ready - as Im trying to clear our the queue.

Hello @hectorkambow , on my end I think ok. Unless there are more changes that will be requested by the Peer reviewer.

**Comment by lalith1403** [comment]
> @OMONDIELLY 
> 
> 
> 
> cc @lalith1403  
> 
> 
> 
> Just wanted to get a status update on this PR - see if it was ready - as Im trying to clear our the queue.

Will perform the tech review today. Thank you.

File: content/articles/working-with-styled-components-in-react/index.md [hunk 1]
```diff
@@ -0,0 +1,464 @@
...
+2. Basic knowledge of [JavaScript programming language](https://www.w3schools.com/js/DEFAULT.asp) and [CSS](https://www.w3schools.com/css/).
+
+3. A package manager, [yarn](https://yarnpkg.com/) or [npm](https://www.npmjs.com/get-npm).
+
+4. Code Editor of your choice. I will be using [VS Code](https://code.visualstudio.com/download).
+
+5. [Node.js](https://www.nodejs.org). You can download the latest version.
+
+
+If you have worked with React in developing web applications, you will agree with me that adding style to your Components is super cool when using the styled-components feature. In any case, you have not used React or styled-components, do not worry because this article is for you.
```
Comment (lalith1403) [inline content/articles/working-with-styled-components-in-react/index.md:32]: In any case, if you have not used React or styled-components, do not worry as this article is for you.

File: content/articles/working-with-styled-components-in-react/index.md [hunk 2]
```diff
@@ -0,0 +1,464 @@
...
+This article will cover:
+- [What is React?](#What-is-react)
+- [What is Styled-components](#What-is-Styled-components)
+- [Advantages of using Styled-components](#Advantages-of-using-styled-components)
+- [Creating and styling a simple web-page using styled-components](#Styling-a-simple-webpage)
+
+
+
+### What is React
+React is a JavaScript library built and maintained by Facebook. React is purely Open-source.
```
Comment (lalith1403) [inline content/articles/working-with-styled-components-in-react/index.md:48]: Change the second sentence, purely open-source. Insert it in the first sentence as: React is an open-source JavaScript library built and maintained by Facebook.

File: content/articles/working-with-styled-components-in-react/index.md [hunk 3]
```diff
@@ -0,0 +1,464 @@
...
+When faced with the need to build elegant and responsive user interfaces/UI components, React is the go-to tool.
```
Comment (lalith1403) [inline content/articles/working-with-styled-components-in-react/index.md:49]: Go-to framework

File: content/articles/working-with-styled-components-in-react/index.md [hunk 4]
```diff
@@ -0,0 +1,464 @@
...
+React is preferred by Front End developers because it is Component-Based. 
+
+React as a frontend library integrates well with backend development libraries such as Node.js, Django, Java, and Ruby.
+React is fast, simple, and scalable because data changes in your applications can be rendered without requiring you to reload your application.
+
+Each part of your application in React is built separately and reusably as a [Component](https://www.w3schools.com/react/react_components.asp) with the component-based feature. The components are then composed together to form complex user interfaces. In addition, components allow developers to split their UIs into independent and reusable sections to engage each section in isolation.
```
Comment (lalith1403) [inline content/articles/working-with-styled-components-in-react/index.md:55]: This needs further refining. The concept behind React is modular components. You should consider explaining it in a way such that the concept of modularity is a consequence of the design principles of React.

File: content/articles/working-with-styled-components-in-react/index.md [hunk 5]
```diff
@@ -0,0 +1,464 @@
...
+
+Components in React can be written as Functions or Classes, all of which accept defined inputs as [props](https://reactjs.org/docs/components-and-props.html) to determine what is rendered on the browsers.
```
Comment (lalith1403) [inline content/articles/working-with-styled-components-in-react/index.md:57]: Further refining is required. What is the utility of functional components and class based components. Define props formally.

File: content/articles/working-with-styled-components-in-react/index.md [hunk 6]
```diff
@@ -0,0 +1,464 @@
...
+  return 
+     (<Container>
+         <Content>
+         <Content>
+     </Container>);
+}
+export default Landing;
+```
+
+To create our second component, open the *Header.js* file that is in the components folder and add the following code:
```
Comment (lalith1403) [inline content/articles/working-with-styled-components-in-react/index.md:159]: Open the Header.js file in the components folder. Add the following code to the file.

File: content/articles/working-with-styled-components-in-react/index.md [hunk 7]
```diff
@@ -0,0 +1,464 @@
...
+    </Nav>
+    );
+}
+
+export default Header;
+```
+
+#### Routing a Component into the main App
+
+To get things started, let's open *App.js*  which will be the base of our application, and replace all the content in it with the code below to create an App function that will load our styled-components and render them as the landing page.
```
Comment (lalith1403) [inline content/articles/working-with-styled-components-in-react/index.md:179]: Introduce react-router library briefly.

File: content/articles/working-with-styled-components-in-react/index.md [hunk 8]
```diff
@@ -0,0 +1,464 @@
...
+
+```JavaScript
+import {BrowserRouter as Router, Switch, Route} from "react-router-dom";//The Router, Switch, and Route will help us move between our created component and the main App.js.
+Import Landing from "./components/Landing" //This is to import the component created in the Landing.js file.
```
Comment (lalith1403) [inline content/articles/working-with-styled-components-in-react/index.md:183]: Check your import statements.

File: content/articles/working-with-styled-components-in-react/index.md [hunk 9]
```diff
@@ -0,0 +1,464 @@
...
+ <Landing/>
+```
+```JavaScript
+ <Header/>
+```
+The final *App.js* should now contain the following code:
+
+```JavaScript
+
+import {BrowserRouter as Router, Switch, Route} from "react-router-dom";
```
Comment (lalith1403) [inline content/articles/working-with-styled-components-in-react/index.md:215]: Briefly explain why react-router-dom is needed.

File: content/articles/working-with-styled-components-in-react/index.md [hunk 10]
```diff
@@ -0,0 +1,464 @@
...
+```
+
+Next, add the code below between the *Nav tags* in the Header component.
+
+```JavaScript
+        <Logo>
+             <img src="/images/logo.svg" alt= "Disney+"/>
+        </Logo> 
+```
+*Bravo if you reached up to this point!*
```
Comment (lalith1403) [inline content/articles/working-with-styled-components-in-react/index.md:270]: Add a screenshot of the expected output at this stage. Do the same for various stages previously that you seem fit.

File: content/articles/working-with-styled-components-in-react/index.md [hunk 11]
```diff
@@ -0,0 +1,464 @@
...
+const Container = styled.section`  
+overflow: hidden;
+display: flex;
+flex-direction: column;
+text-align: center;
+height: 100vh;
+`;
+```
+
+#### Styling the Content Area
```
Comment (lalith1403) [inline content/articles/working-with-styled-components-in-react/index.md:289]: Explain what you are trying to do with styling. For example, The image needs to be center aligned. Therefore, I am using the following CSS properties. Flex display helps me achieve this along with other advantages.

This is the template you should follow while explaining CSS files. Why are you using the properties that you are using. Explain that.

**Review by lalith1403 — CHANGES_REQUESTED** [review]
Fantastic article. A few minor changes and additions, and this should be good to go.

**Comment by ninjaginja** [comment]
plagiarism check completed - all good
[2120420-5ba3b6ca379c9aeaaa7d-1626730383642.pdf](https://github.com/section-engineering-education/engineering-education/files/6844038/2120420-5ba3b6ca379c9aeaaa7d-1626730383642.pdf)

**Comment by OMONDIELLY** [comment]
> Fantastic article. A few minor changes and additions, and this should be good to go.
Hello @lalith1403 , I have made the requested changes.

File: content/articles/working-with-styled-components-in-react/index.md [hunk 12]
```diff
@@ -0,0 +1,504 @@
...
+
+export default App;
+```
+![App.js final](/engineering-education/working-with-styled-components/App.js.png)
+
+### Onto some styling now
+
+Be sure to create an images folder inside the public folder of the application.
+
+You should have the landing page background image, disney+ icon image, logo-one, and logo-two within the folder.
```
Comment (lalith1403) [inline content/articles/working-with-styled-components-in-react/index.md:257]: Add links to the images that you have used. That would be better than me going to search for images on the site mentioned.

File: content/articles/working-with-styled-components-in-react/index.md [hunk 13]
```diff
@@ -0,0 +1,504 @@
...
+}
+
+export default Header;
+```
+
+#### Routing a Component into the main App
+
+To get things started, let's open `App.js`  which is the base of our application. We replace all the content in it with the code below to create an App function that will load our styled-components and render them as the landing page.
+
+After building components/pages in your web application, you may need to expose and let your users navigate through them. To achieve this, you need a dedicated router. [React Router](https://www.freecodecamp.org/news/react-router-cheatsheet/) is a suite of navigational packages/components such as react-router-dom.
```
Comment (lalith1403) [inline content/articles/working-with-styled-components-in-react/index.md:191]: Explain react-router intuitively. You may direct the reader to an in-depth article on react-router. Navigational components such as react-router-dom does not convey information

**Review by lalith1403 — CHANGES_REQUESTED** [review]
2 minor improvements and this should be good to go.

**Comment by lalith1403** [comment]
@OMONDIELLY Ping me once you are done with the changes.

**Comment by OMONDIELLY** [comment]
> @OMONDIELLY Ping me once you are done with the changes.

Sure.. Thanks

**Comment by OMONDIELLY** [comment]
> @OMONDIELLY Ping me once you are done with the changes.

@lalith1403 , Kindly check the updates

**Comment by hectorkambow** [comment]
@OMONDIELLY 
Please be sure to go over my final edits to improve future articles and reviews. 

I believe you were interested in becoming a peer reviewer - these edits are things you will be trying to fix for other students or help them prevent these mistakes.

cc @lalith1403 please do the same - some edits to the front matter and through out the article

**Comment by OMONDIELLY** [comment]
> @OMONDIELLY
> Please be sure to go over my final edits to improve future articles and reviews.
> 
> I believe you were interested in becoming a peer reviewer - these edits are things you will be trying to fix for other students or help them prevent these mistakes.
> 
> cc @lalith1403 please do the same - some edits to the front matter and through out the article

Following the edits keenly. Thanks.

**Pull request merged**